libdrgn/stack_trace.c

```diff
@@ -1106,9 +1114,25 @@ drgn_object_stack_trace(const struct drgn_object *obj,
 		if (err)
 			return err;
 		return drgn_get_stack_trace(drgn_object_program(obj),
-					    value.uvalue, NULL, ret);
+					    value.uvalue, NULL, NULL, ret);
 	} else {
-		return drgn_get_stack_trace(drgn_object_program(obj), 0, obj,
+		return drgn_get_stack_trace(drgn_object_program(obj), 0, obj, NULL,
 					    ret);
 	}
 }
+
+LIBDRGN_PUBLIC struct drgn_error *
+drgn_thread_stack_trace(struct drgn_thread *thread,
+			struct drgn_stack_trace **ret)
+{
+	if (thread->prog->flags & DRGN_PROGRAM_IS_LINUX_KERNEL) {
+		struct nstring *prstatus =
+			!(thread->prog->flags & DRGN_PROGRAM_IS_LIVE) &&
```

Comment:
This `IS_LIVE` check is redundant here, since we'll set `prstatus.str` to `NULL` for live threads anyways, right?

Reply:
Not quite. I wanted to follow the existing semantics of `drgn_get_stack_trace` and `drgn_get_initial_registers` (i.e. `NULL` is passed for the value of an optional parameter to indicate its absence), hence the check and the subseqent passing of a `NULL` pointer directly instead of a `prstatus` with a `NULL` `.str` member.